spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/apache/RetryableRibbonLoadBalancingHttpClient.java

```diff
@@ -132,6 +133,10 @@ public RibbonApacheHttpResponse doWithRetry(RetryContext context) throws Excepti
 							.port(service.getPort()).path(newRequest.getURI().getPath())
 							.query(newRequest.getURI().getQuery()).fragment(newRequest.getURI().getFragment())
 							.build().encode().toUri());
+					
+					if (service instanceof RibbonServer) {
+						statsRecorder = new RibbonStatsRecorder(ribbonLoadBalancerContext, ((RibbonServer)service).getServer());
```

Comment:
is it ok if `ribbonLoadBalancerContext` is null?  We might need a null check here because there is no guarantee the setter is called